drivers/input/input_vs1838b.c

```diff
@@ -315,6 +315,15 @@ static void vs1838b_input_callback(struct device const *dev, struct gpio_callbac
 		data->edges_ticks[data->edges_count++] = tick;
 	}
 
+	/* If the first 3 edges do not match a leading burst,
+	 * shift left the edges_ticks to get rid of leading noises.
+	 */
+	if ((data->edges_count == 3) && !detect_leading_burst(data->edges_ticks)) {
```

Comment:
Maybe replace 3 with `(NEC_LEAD_PULSE_EDGE_WIDTH + 1)` to make clear where the 3 comes from.